src/hotspot/share/jvmci/jvmciEnv.hpp

```diff
@@ -146,6 +146,7 @@ class JVMCICompileState : public ResourceObj {
   void inc_compilation_ticks();
 };
 
+extern JNIEXPORT jobjectArray c2v_getDeclaredFieldsInfo(JNIEnv* env, jobject, jobject, jlong);
```

Comment:
What's the purpose of this declaration? I don't think you need it or the `friend` declaration below since `new_FieldInfo(FieldInfo* fieldinfo, JVMCI_TRAPS)` is public.

Reply:
Without this declaration, builds fail on Windows with this error:
`error C2375: 'c2v_getDeclaredFieldsInfo': redefinition; different linkage`

Reply:
Strange - thats not needed for other `JVMCIEnv` methods called from `jvmciCompiler.ToVM.cpp`. There must be some way to avoid this.

Reply:
The issue was caused by the `friend` declaration below (I cannot remember why I added in the first place), which seems to add an implicit declaration of the method that was conflicting with the original declaration of the method. Once the `friend` declaration is removed, builds on Windows don't need the `extern` declaration anymore.